src/fork_gc.c

```diff
@@ -81,14 +81,21 @@ static void FGC_updateStats(RedisSearchCtx *sctx, ForkGC *gc, size_t recordsRemo
 
 static void FGC_sendFixed(ForkGC *fgc, const void *buff, size_t len) {
   RS_LOG_ASSERT(len > 0, "buffer length cannot be 0");
-  ssize_t size = write(fgc->pipefd[GC_WRITERFD], buff, len);
-  if (size != len) {
-    if (size == -1) {
-      perror("write()");
-      abort();
-    } else {
-      RS_LOG_ASSERT(size == len, "buffer failed to write");
+  ssize_t size = 0;
+  while (size < len) {
```

Comment:
I think that the while is not needed, our pipe is in blocking mode, which means that it will block until there is enough space to write all the data. If you get -1 or a size that is less than the size you wrote it means that the pipe is close.